src/tools/wasm-interp.cc

```diff
@@ -239,6 +346,37 @@ static Result ReadAndRunModule(const char* module_filename) {
 
   RefVec imports;
 
+  // if we only have dummy imports, we wont need to reghister anything
+  // but we still want to load them.
+  std::vector<Module::Ptr> modules_loaded(s_modules.size());
```

Comment:
Can we unify the loading of the main `module_filename` above with the loading of `s_modules` here?

Reply:
I was thinking about it early, but dropped it, because I did not understand how it would make it work.
Probably why I have that vector in the first place.
Now on my TODO. After I figure out how to do the module option thingy.

Reply:
Should I add a Divider to Option Parser?
```cpp
  parser.AddArgument("filename", OptionParser::ArgumentCount::OneOrMore, [...]);
  // parser.AddDivider("--", OptionParser::Divider::Optional);
  parser.AddArgument("arg", OptionParser::ArgumentCount::ZeroOrMore, [...]);
```
This way I could distinguish between modules and wasi arguments.

Reply:
Yes, I think using `--` to distinguish between arguments passed to wasi and argument to wasm-interp itself is good idea.

In general I'm a fan of the direction of this PR, and I think we could probably even land with without too much more iteration, but I would also like to be able to iterate on the ergonomics of it.  But perhaps we can do a lot of that iteration after first landing it.

Reply:
Well: Off I go to find out how to implement that... Possibly wihtout implementing a whole Divider thing.